packages/frontend-core/src/components/grid/cells/RelationshipCell.svelte

```diff
@@ -29,17 +29,35 @@
   let searching = false
   let container
   let anchor
+  let cacheStr
 
+  $: fieldValue = parseValue(value)
   $: oneRowOnly = schema?.relationshipType === "one-to-many"
   $: editable = focused && !readonly
-  $: lookupMap = buildLookupMap(value, isOpen)
+  $: lookupMap = buildLookupMap(fieldValue, isOpen)
   $: debouncedSearch(searchString)
   $: {
     if (!focused && isOpen) {
       close()
     }
   }
 
+  const parseValue = value => {
+    // Is it unset or a valid array? 1+
+    const isValid = val =>
+      !val ||
+      (Array.isArray(val) &&
+        (val.length === 0 ||
+          val.some(relEntry => Object.hasOwn(relEntry, "_id"))))
+
+    const stf = JSON.stringify(value)
+    if (!cacheStr || cacheStr !== stf) {
+      cacheStr = stf
```

Comment:
What is the logic behind this cache? Just not to call the isValid?

Reply:
I ran into some issues with reactivity at one point and was caching to help debug. As @aptkingston, this can be made way simpler.